src/plasma/plasma_store.cc

```diff
@@ -340,33 +239,37 @@ void return_from_get(PlasmaStoreState *store_state, GetRequest *get_req) {
    * tables if it is present there. It should only be present there if the get
    * request timed out. */
   for (ObjectID &object_id : get_req->object_ids) {
-    remove_get_request_for_object(store_state, object_id, get_req);
+    auto &get_requests = object_get_requests_[object_id];
+    /* Erase get_req from vector, see http://stackoverflow.com/a/16013546 */
```

Comment:
I see what you're doing here now. This idiom is used to remove *all* instances of `get_req` from the `get_requests` vector. However, since `get_req` can only occur once in the vector (we were relying on this fact before), this could also be done with 

```
get_requests.erase(std::find(get_requests.begin(), get_requests.end(), get_req))
```

which is probably a bit cleaner.